plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -844,25 +844,49 @@ func (checker *PodAffinityChecker) InterPodAffinityMatches(pod *api.Pod, meta in
 	if err != nil {
 		return false, err
 	}
-	if checker.NodeMatchPodAffinityAntiAffinity(pod, allPods, node) {
-		return true, nil
+	affinity, err := api.GetAffinityFromPodAnnotations(pod.Annotations)
```

Comment:
The rest of this function is basically a copy from: "NodeMatchPodAffinityAntiAffinity" below.